docs/source/prjxray/vtr-rrgraph.rst

```diff
@@ -0,0 +1,408 @@
+
+.. thumbnail:: ../images/prjxray/import-flow.png
+
+   This diagram shows the importing flow for Project X-Ray.
+
+Tile wire classification
+------------------------
+
+Before channels can be formed, **tile wires** need to be bucketed into
+their purpose.
+
+Step (1) - Group tile wires into “nodes”
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+   The first step is to first re-form nodes that contain all the
+   directly connected **tiles wires** These nodes are *not* VPR
+   **nodes**, they are simply the collection of **tile wires** that are
+   already a net (electrically equivalent).
+
+Step (2) - Classify “nodes”
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+   Each node then needs to be classified. The simplest classification is
+   a channel wire, which means that **pips** route on and off of the
+   node. However there are other important classifications. For example,
+   the carry chain connection between two CLBLL_L tiles should be
+   modelled as a tile direct connection, rather than routing onto a
+   channel and back off. The is classified as a "edge with mux".
+
+The classification is broken down into the following categories:
+
+-  CHANNEL - **Pips** route on and off of this node.
+
+-  EDGE_WITH_MUX - Two **tile wires** connected by a **pip**.
+
+   -  The first **tile wire** sources at a **site pin**, and the second
+      **tile wire** sinks at a **site pin**.
+
+   -  This captures direct inter-tile connections like carry chain
+      wires, BRAM data cascade wires, etc.
+
+-  NULL - A node that has either no source or no sink. This wires
+   typically occur near the edge of the grid.
+
+-  EDGES_TO_CHANNEL - A node that sources and sinks from a **site pin**
+   and connects via a **pip** to a CHANNEL
+
+.. figure:: ../images/prjxray/import-wire-class.svg
+
+
+There is another classification EDGE_WITH_SHORT, which is a direct
+connection between two **site pins**. This does not appear to occur in
+7-series parts.
+
+The reason this classification is important is that each node that is a
+CHANNEL must be mapped into one or more CHANX or CHANY **nodes**.
+EDGE_WITH_MUX nodes must be converted into root level architecture
+direct connections, and will be **edges** between two **site pin**
+nodes. EDGES_TO_CHANNEL will be become **edges** in the routing between
+**site pins nodes** and CHANX/CHANY **nodes**.
+
+Channel formation
+-----------------
+
+All nodes that were classified as CHANNEL type need to assigned CHANX
+and CHANY dimensions. This is done via
+`make_tracks <https://github.com/SymbiFlow/symbiflow-arch-defs/blob/master/utils/lib/rr_graph/tracks.py>`_.
+make_tracks takes a point bag containing all of the source and sink grid
+locations for a particular channel. It returns straight lines such that
+all sources and sink grid locations can route on to or off of the
+channel.
+
+Point Bag to CHANX / CHANY decomposition
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. figure:: ../images/prjxray/vtr-rrgraph.png
```

Comment:
This figure spills in the ToC? https://symbiflow--1365.org.readthedocs.build/projects/arch-defs/en/1365/prjxray/vtr-rrgraph.html

Reply:
Now it should be fine